src/quokka_editor_back/routers/documents.py

```diff
@@ -118,3 +99,17 @@ async def delete_document(
 #     )
 #     await document.save()
 #     return document
+
+
+@router.post("/share/{document_id}")
+async def share_document(
+    document_id: UUID,
+    current_user: Annotated[User, Depends(get_current_user)],
+):
+    await get_document(document_id=document_id, user=current_user)
+
+    token = create_access_token(
+        data={"document_id": str(document_id), "permissions": "edit"}
```

Comment:
Should there always be edit permission? What with View permission?

Reply:
I forgot to add more permissions here.